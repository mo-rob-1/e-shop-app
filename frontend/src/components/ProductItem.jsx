import { Link } from "react-router-dom";

function ProductItem({ imageUrl, description, price, name, productId }) {
  return (
    <div className="shadow-md rounded-bl-xl rounded-br-xl">
      <img
        src={imageUrl}
        alt={name}
        height="300px"
        className="rounded-tr-xl rounded-tl-xl w-full h-80 object-cover"
        data-testid="product-item-image"
      />

      <div className="pl-6 pr-6 pt-8 pb-8">
        <h2 className="font-bold text-lg mb-4" data-testid="product-item-name">
          {name}
        </h2>

        <p className="mb-3" data-testid="product-item-desc">
          {description?.substring(0, 100)}...
        </p>

        <p className="mb-3 font-bold text-2xl" data-testid="product-item-price">
          Â£{price}
        </p>

        <Link
          className="bg-slate-800 text-white font-bold inline-block pl-10 pr-10 pt-4 pb-4 mt-2 text-sm rounded-md"
          to={`/product/${productId}`}
          data-testid="product-item-view-link"
        >
          View
        </Link>
      </div>
    </div>
  );
}

export default ProductItem;
