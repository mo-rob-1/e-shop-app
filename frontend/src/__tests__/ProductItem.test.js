import { render } from "@testing-library/react";
import ProductItem from "../components/ProductItem";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import store from "../redux/store";
import { BrowserRouter as Router } from "react-router-dom";

describe("ProductItem", () => {
  it("renders the product item", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Provider store={store} history={history}>
        <Router>
          <ProductItem />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the product item image", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <ProductItem />
        </Router>
      </Provider>
    );

    const productItemImage = getByTestId("product-item-image");
    expect(productItemImage).toBeTruthy();
  });

  it("renders the product item name", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <ProductItem />
        </Router>
      </Provider>
    );

    const productItemName = getByTestId("product-item-name");
    expect(productItemName).toBeTruthy();
  });

  it("renders the product item description", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <ProductItem />
        </Router>
      </Provider>
    );

    const productItemDescription = getByTestId("product-item-desc");
    expect(productItemDescription).toBeTruthy();
  });

  it("renders the product item price", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <ProductItem />
        </Router>
      </Provider>
    );

    const productItemPrice = getByTestId("product-item-price");
    expect(productItemPrice).toBeTruthy();
  });

  it("renders the view product link", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <ProductItem />
        </Router>
      </Provider>
    );

    const viewProductItemLink = getByTestId("product-item-view-link");
    expect(viewProductItemLink).toBeTruthy();
  });
});
