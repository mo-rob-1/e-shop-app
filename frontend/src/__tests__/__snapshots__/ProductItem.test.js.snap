// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem renders the product item 1`] = `
<DocumentFragment>
  <div
    class="shadow-md rounded-bl-xl rounded-br-xl"
  >
    <img
      class="rounded-tr-xl rounded-tl-xl w-full h-80 object-cover"
      data-testid="product-item-image"
      height="300px"
    />
    <div
      class="pl-6 pr-6 pt-8 pb-8"
    >
      <h2
        class="font-bold text-lg mb-4"
        data-testid="product-item-name"
      />
      <p
        class="mb-3"
        data-testid="product-item-desc"
      >
        ...
      </p>
      <p
        class="mb-3 font-bold text-2xl"
        data-testid="product-item-price"
      >
        $
      </p>
      <a
        class="bg-slate-800 text-white font-bold inline-block pl-10 pr-10 pt-4 pb-4 mt-2 text-sm rounded-md"
        data-testid="product-item-view-link"
        href="/product/undefined"
      >
        View
      </a>
    </div>
  </div>
</DocumentFragment>
`;
